operator/src/main/java/org/bf2/operator/managers/MetricsManager.java

```diff
@@ -0,0 +1,121 @@
+package org.bf2.operator.managers;
+
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.informers.ResourceEventHandler;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.search.Search;
+import io.quarkus.runtime.Startup;
+import io.strimzi.api.kafka.KafkaList;
+import io.strimzi.api.kafka.model.Kafka;
+import io.strimzi.api.kafka.model.listener.arraylistener.GenericKafkaListener;
+import org.bf2.common.ResourceInformer;
+import org.bf2.common.ResourceInformerFactory;
+import org.bf2.operator.operands.KafkaCluster;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import java.util.Locale;
+import java.util.stream.IntStream;
+
+import static org.bf2.operator.managers.InformerManager.filter;
+import static org.bf2.operator.operands.AbstractKafkaCluster.EXTERNAL_LISTENER_NAME;
+
+@Startup
+@ApplicationScoped
+public class MetricsManager {
+    static final String KAFKA_INSTANCE_BROKERS_DESIRED_COUNT = "kafka_instance_brokers_desired_count";
+    static final String KAFKA_INSTANCE_PARTITION_LIMIT = "kafka_instance_partition_limit";
+    static final String KAFKA_INSTANCE_MAX_MESSAGE_SIZE_LIMIT = "kafka_instance_max_message_size_limit";
+    static final String KAFKA_INSTANCE_CONNECTION_LIMIT = "kafka_instance_connection_limit";
+    static final String KAFKA_INSTANCE_CONNECTION_CREATION_RATE_LIMIT = "kafka_instance_connection_creation_rate_limit";
+
+    static final String TAG_LABEL_OWNER = "owner";
+    static final String TAG_LABEL_BROKER_ID = "broker_id";
+    static final String TAG_LABEL_NAMESPACE = "namespace";
+    static final String TAG_LABEL_INSTANCE_NAME = "instance_name";
+
+    static final Tag OWNER = Tag.of(TAG_LABEL_OWNER, "KafkaInstanceMetricsManager");
+    static final String TAG_LABEL_LISTENER = "listener";
+
+    @Inject
+    ResourceInformerFactory resourceInformerFactory;
+
+    @Inject
+    MeterRegistry meterRegistry;
+
+    @Inject
+    KubernetesClient kubernetesClient;
+
+    private volatile ResourceInformer<Kafka> kafkaResourceInformer;
+
+    protected synchronized void createInformer() {
+        if (kafkaResourceInformer != null) {
+            return;
+        }
+        kafkaResourceInformer = resourceInformerFactory.create(Kafka.class, filter(kubernetesClient.resources(Kafka.class, KafkaList.class)), new ResourceEventHandler<>() {
```

Comment:
With some refactoring we should be able to this handler to the existing informer, so that we don't have to maintain a separate cache/watch.

Reply:
I spent a while looking at whether i could layer responsibilities into `ManagedKafkaController`, but I didn't really see the way forward.  I'm not too familiar with code, so I could well be missed better solutions.  I do like the separation of concerns that the current solution brings.

I'm open to refactoring what I have but I'll need some guidance.

Reply:
@shawkins I struggled a fair bit  with this refactoring.  The difficulty was that the kafka informer is created late (only once the operate knows the CRDs are present).  That meant I needed some mechanism to trigger the registration of the handler with the kafka informer, which lead me to `org.bf2.common.ResourceInformerFactory#addEventHandler`. I'm okay with the production code.

I'm not so happy with MockResourceInformerFactory - I've made the code worse.  I think MockResourceInformerFactory would be better eliminated, but some of the existing tests seem to rely on subtleties of the mock's behaviour (IngressControllerManagerTest).   Refactoring the tests that quickly got beyond what I was happy with for this PR, so I would prefer remove the dependency on a separate PR. 

I did try some alternative paths: using `QuarkusMock.installMockForType` for ResourceInformerFactory, but it seems to get confused by the presence of the MockResourceInformerFactory. 

WDYT?

Reply:
> I'm not so happy with MockResourceInformerFactory - I've made the code worse.

My original intention was to disable all async event processing - whether scheduled or something that would be handled by an informer, so that tests could be written based upon just single threaded behavior.  

The addition of:

if (eventHandler != null) {
   watchListDeletable.inform((ResourceEventHandler) eventHandler);
}

Definitely violates that as that is starting up a real informer - I misread the pr initially and should have said something earlier.

I don't mind refactoring to remove MockResourceInformerFactory entirely.  The intermediate step is to make sure the mock server state is reset as necessary.

But the goal of no-async behavior is very difficult when starting a full quarkus - beans may be being started in parallel and if they are all making updates to the same mock store, then you may see things you don't expect.  When we upgrade quarkus we should be able to remove the startup annotation from a lot of bean - lazy initialization will make things a bit better.